### PR TITLE
Created integration test for WebConnectivity and API

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Part of https://github.com/ooni/backend/issues/450

### CODE REVIEW

File: app/src/androidTest/java/org/openobservatory/ooniprobe/test/WebConnectivityTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+public class WebConnectivityTest extends AbstractTest {
+    private static final String MK_NAME = "WebConnectivity";
+    private static final String TAG = "integration-test";
+    private static final String UNUSED_KEY = "UNUSED_KEY";
+    private static final String CLIENT_URL = "https://ams-pg.ooni.org";
+
+    @Test
+    public void runWebConnectivity() {
+        final CountDownLatch signal = new CountDownLatch(1);
```
Comment (bassosimone) [inline app/src/androidTest/java/org/openobservatory/ooniprobe/test/WebConnectivityTest.java:27]: This code is synchronous, so I think we don't need to use a countdown latch here.

File: app/src/androidTest/java/org/openobservatory/ooniprobe/test/WebConnectivityTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import org.openobservatory.ooniprobe.AbstractTest;
+import org.openobservatory.ooniprobe.model.jsonresult.EventResult;
+import org.openobservatory.ooniprobe.model.jsonresult.JsonResult;
+import org.openobservatory.ooniprobe.model.settings.Settings;
+
+import java.util.Collections;
+import java.util.concurrent.CountDownLatch;
+
+public class WebConnectivityTest extends AbstractTest {
+    private static final String MK_NAME = "WebConnectivity";
```
Comment (bassosimone) [inline app/src/androidTest/java/org/openobservatory/ooniprobe/test/WebConnectivityTest.java:20]: Maybe call this `EXPERIMENT_NAME`?

**Review by bassosimone — CHANGES_REQUESTED** [review]

**Comment by bassosimone** [comment]
What is missing:

- [ ] the API returns correct URL
- [ ] we run this test periodically using GHA

Regarding the latter, @lorenzoPrimi, I recommend checking how we do that in https://github.com/ooni/e2etesting/blob/master/.github/workflows/debian9.yml

File: app/src/androidTest/java/org/openobservatory/ooniprobe/client/OONIAPIClientTest.java [hunk 3]
```diff
@@ -25,16 +25,28 @@
     private static final String EXISTING_REPORT_ID = "20190113T202156Z_AS327931_CgoC3KbgM6zKajvIIt1AxxybJ1HbjwwWJjsJnlxy9rpcGY54VH";
     private static final String NONEXISTING_REPORT_ID = "EMPTY";
     private static final String NON_EXISTING_MEASUREMENT_URL = "https://api.ooni.io/api/v1/measurement/nonexistent-measurement-url";
```
[the remark this answers is not in the document]
Comment (lorenzoPrimi) [reply app/src/androidTest/java/org/openobservatory/ooniprobe/client/OONIAPIClientTest.java:27]: In iOS we use `https://\t`

**Review by bassosimone — APPROVED** [review]
🐳

**Pull request merged**